packages/fiber/src/core/utils.ts

```diff
@@ -23,6 +23,14 @@ export function calculateDpr(dpr: Dpr) {
   return Array.isArray(dpr) ? Math.min(Math.max(dpr[0], window.devicePixelRatio), dpr[1]) : dpr
 }
 
+// Prunes or picks keys from an object
+export function filterKeys<O extends { [key: string]: any }, K extends string[]>(obj: O, prune = false, ...keys: K) {
+  // @ts-ignore
+  const keysToSelect = new Set(keys.flat())
+  // @ts-ignore
+  return Object.fromEntries(Object.entries(obj).filter(([key]) => keysToSelect.has(key) === !prune))
+}
+
```

Comment:
Is there a way to type the result based on whether it's pruning or picking?

Also need to resolve TS and the usage of newer web APIs here.

Reply:
I think this might be better suited... although it's not perfect as I'm not full clear what `prune` should do...

```js
export function filterKeys<
  TObj extends { [key: string]: any },
  TPrune extends boolean,
  TKey extends keyof TObj
>(
  obj: TObj,
  prune: TPrune,
  ...keys: TKey[]
): TPrune extends true ? Omit<TObj, TKey> : Pick<TObj, TKey> {
  // @ts-ignore
  const keysToSelect = new Set(keys.flat())
  // @ts-ignore
  return Object.fromEntries(Object.entries(obj).filter(([key]) => keysToSelect.has(key) === !prune))
}
```

Reply:
I see. Prune has this toggle between omitting and picking. Can probably be better named.

Reply:
Did that work?

Reply:
It did. I just created pick and omit methods to mirror lodash's. Had to drop those since they were modules and broke tests.

Thought I could borrow this util from react-ogl.